src/panel/direction/MobileRoadMapPreview.jsx

```diff
@@ -2,62 +2,79 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import RoadMapStep from './RoadMapStep';
 import { fire } from 'src/libs/customEvents';
-import { CloseButton } from 'src/components/ui';
 
 export default class MobileRoadMapPreview extends React.Component {
   static propTypes = {
     steps: PropTypes.array.isRequired,
     onClose: PropTypes.func.isRequired,
-  }
+  };
 
-  state = {
-    currentStep: 0,
-  }
+  state = { };
 
   componentDidMount() {
-    this.zoomToCurrentStep();
-    fire('move_mobile_geolocation_button', 65);
+    this.steps = document.querySelector('.mobile-roadmap-preview-steps');
+    this.stepList = document.querySelectorAll('.itinerary_mobile_step');
+    this.currentStep = 0;
+    fire('move_mobile_bottom_ui', 105);
+    this.highlightCurrentStep();
   }
 
-  zoomToCurrentStep = () => {
-    fire('zoom_step', this.props.steps[this.state.currentStep]);
-  }
+  highlightCurrentStep = () => {
```

Comment:
This follows my previous comment: most of what you do imperatively in this method is the job of React.
By managing `currentStep` in the state, each change to it will trigger a re-render. In the `render` method, you'll have all the necessary pieces to assign the right `className` to each step DOM element while you render them, without the need for an external loop like this.

The main thing to keep in mind: in React, we don't render once then manipulate the DOM. Instead, we manipulate data state and the DOM always follows by reflecting changes.